File: rabbitmq-pub/publish.rb
#!/usr/bin/env ruby
# encoding: utf-8
 
require "rubygems"
require "amqp"
 
EventMachine.run do
  connection = AMQP.connect(:host => 'localhost', :port => 5001)
  puts "Connected to AMQP broker. Running #{AMQP::VERSION} version of the gem..."
 
  channel  = AMQP::Channel.new(connection)
  queue    = channel.queue("amqpgem.examples.helloworld")
  exchange = channel.direct("")

  exchange.publish "Hello, world!", :routing_key => queue.name
end	

File: rabbitmq-pub/publish.py
import pika
import sys

connection = pika.BlockingConnection(pika.ConnectionParameters(
        host='localhost', port=5001))
channel = connection.channel()

channel.exchange_declare(exchange='jspyconf.helloworld2',
                         type='fanout')

message = "Hello World!"
channel.basic_publish(exchange='jspyconf.helloworld2',
                      routing_key='',
                      body=message)
print " [x] Sent %r" % (message,)
connection.close()